test/functional/discovered_hosts_controller_test.rb

```diff
@@ -127,6 +127,7 @@ def test_edit_form_submit_parameters
   end
 
   def test_update_inheritance
+    User.current = User.unscoped.find(set_session_user_default_manager[:user])
```

Comment:
Can you refactor the helpers and make new `user_default_manager` and use it in `set_session_user_default_manager`? The same for reader, this is little bit clunky :-)